calculation/calculation.go

```diff
@@ -142,21 +149,29 @@ func CalculateTotalLPAtSnapshot(
 		if !activeAtSnapshot {
 			continue
 		}
-		for assetId, amount := range position.Value.Assets {
-			if poolLookup.IsLPToken(assetId) {
-				pool, err := poolLookup.PoolByLPToken(ctx, assetId)
-				if err != nil {
-					return nil, nil, nil, 0, fmt.Errorf("failed to lookup pool for LP token %v: %w", assetId, err)
-				}
-				poolsByIdent[pool.PoolIdent] = pool
-				lockedLPByIdent[pool.PoolIdent] += amount.Uint64()
-				if pool.AssetA == "" {
-					lockedLP := amount.BigInt()
-					totalLP := num.Uint64(pool.TotalLPTokens).BigInt()
-					numerator := big.NewInt(0).Mul(lockedLP, num.Uint64(pool.AssetAQuantity).BigInt())
-					assetA := big.NewInt(0).Div(numerator, totalLP)
-					valueByIdent[pool.PoolIdent] += assetA.Uint64() * 2
-					totalValue += assetA.Uint64() * 2
+
+		for policy, policyMap := range position.Value {
+
+			for _, amount := range policyMap {
+				assetId := shared.AssetID(policy)
```

Comment:
same here

Reply:
I can speak to this one and the other comment. Forrest was running some unit tests that were failing. Apparently, the asset ID in the inner map was blank, which led to `fromSeparate()` generating values like `subaru.`. If the new asset ID was created just from the policy, the tests passed. Perhaps the tests should be revised instead. This was just a way to keep things moving forward.

Reply:
no? that means the implementation of from separate is wrong...

Reply:
In that case, I suppose it should be revised to remove the period if the asset name is empty. I'll patch that right now.

Reply:
> In that case, I suppose it should be revised to remove the period if the asset name is empty. I'll patch that right now.

Just confirm that's actually how Ogmios behaves.

Reply:
I think I just have to update all the AssetIDs in the tests to have a `.`

Reply:
If it turns out ogmios leaves the `.` in, then yes, that would be the appropriate thing to do. Please don't do things *just* to make the tests pass, you first must understand why the test is failing, and whether it's highlighting a problem in the code, or a problem in the test.

Reply:
@malls - I'm guessing you saw my note in Discord, but just in case, I confirmed that, when the asset name is empty, Ogmios just returns the policy name in v5; there's no period. If tests still expect something like "PolicyX." instead of "PolicyX", that need to be fixed.

Reply:
No . expected unless there's an asset name